www/docs/cookbooks/build-go-modules.md

```diff
@@ -0,0 +1,83 @@
+# Building Go modules
+
+With the default configs, you can already build a Go module without issues.
+
+But, if you want to access module information in runtime (e.g. `debug.BuildInfo` or `go version -m $binary`), you'll
+need to setup GoReleaser to "proxy" that module before building it.
+
+To do that, you can simply add this to your config:
+
+```yaml
+# goreleaser.yml
+gomod:
+  proxy: true
+```
+
+In practice, what this does is:
+
+- for each of your builds, create a `dist/proxy/$BUILD_ID`;
+- creates a `go.mod` that requires your root module at the _current tag_;
+- creates a `main.go` that imports your main module;
+- copy the projects `go.sum` to that folder.
+
+In which:
+
+- _root module_: is the output of `go list -m`;
```

Comment:
If you want to align with the Go [module reference](https://golang.org/ref/mod), this is referred to as the main module:

> The main module is the module containing the directory where the go command is invoked.